src/IconButton/InteractiveIcon.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import useActiveElementKeypress from '../hooks/useActiveKeyPress';
+import newId from '../utils/newId';
+
+const iconPrefix = 'interactive-icon';
+
+const InteractiveIcon = React.memo(({
+  altText,
+  darkBackground,
+  icon,
+  iconClassNames,
+  onClick,
+}) => {
+  // if there is a click handler, the icon will act as a button
+  const buttonProps = onClick ? {
+    role: 'button',
+    onClick,
+    tabIndex: 0,
+  } : {};
+
+  /* focusId should be different for each icon generated, all icons on the screen will
+     have click handlers triggerd by spacebar */
+  const focusId = newId(iconPrefix);
+  const darkHoverClass = darkBackground ? 'iconbutton-hover--dark' : '';
+  const darkIconClass = darkBackground ? 'iconbutton--dark' : '';
+
+  const spacePressed = useActiveElementKeypress(focusId, ' ');
+
+  // if the element has a click handler and is therefore acting like a button, the user should
+  // be able to be click it by pressing the space bar
+  if (onClick && spacePressed) {
+    onClick();
+  }
+
+  return (
+    <div id={focusId} className={`iconbutton-hover ${darkHoverClass}`} {...buttonProps}>
```

Comment:
If you use the `button` element pressing space on the keyboard will trigger the onClick (right?). Any reason not to use the button element?

Reply:
Rather than the hook `useActiveElementKeypress` this is probably a better way to reproduce the button spacebar behavior:

```
const handleKeyDown = (e) => {
  if (e.code === 'Space') {
   onClick();
 }
// If a handler is supplied, allow it to handle the event too
props.onKeyDown && props.onKeyDown(e);
};
```
and in the component
```
<div ...
  onKeyDown={handleOnKeyDown}
>
```